torch/_tensor_str.py

```diff
@@ -72,65 +72,52 @@ def __init__(self, tensor):
         self.sci_mode = False
         self.max_width = 1
 
+        # use tensor_view for 0-dim tensor iteration
+        tensor_view = tensor.view(tensor.nelement())
```

Comment:
```suggestion
        with torch.no_grad():
            tensor_view = tensor.view(tensor.nelement())
```

I think this is better, otherwise we might end up having to track autograd graph unnecessarily IIUC.